nodeInterface/NodeInterface.go

```diff
@@ -0,0 +1,434 @@
+// Copyright 2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package nodeInterface
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/offchainlabs/nitro/arbnode"
+	"github.com/offchainlabs/nitro/arbos/retryables"
+	"github.com/offchainlabs/nitro/arbos/util"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/util/arbmath"
+	"github.com/offchainlabs/nitro/util/merkletree"
+	"github.com/offchainlabs/nitro/validator"
+)
+
+// To avoid creating new RPC methods for client-side tooling, nitro Geth's InterceptRPCMessage() hook provides
+// an opportunity to swap out the message its handling before deriving a transaction from it.
+//
+// This mechanism handles messages sent to 0xc8 and uses NodeInterface.sol to determine what to do. No contract
+// actually exists at 0xc8, but the abi methods allow the incoming message's calldata to specify the arguments.
+//
+type NodeInterface struct {
+	Address       addr
+	backend       core.NodeInterfaceBackendAPI
+	context       context.Context
+	sourceMessage types.Message
+	returnMessage struct {
+		message *types.Message
+		changed *bool
+	}
+}
+
+var merkleTopic common.Hash
+var withdrawTopic common.Hash
+
+var blockInGenesis = errors.New("")
+var blockAfterLatestBatch = errors.New("")
```

Comment:
Looks like these should have messages

Reply:
This isn't from my PR but rather is logic carried over from an earlier one. We use `errors.Is` on line 82 which wouldn't work on the variable message text from line 414